### PR TITLE
Add `ConfigBuilder` to build configuration objects

### PR DESCRIPTION
**Pull request by teoljungberg**

### CODE REVIEW

File: app/models/config_builder.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class ConfigBuilder
+  def self.for(hound_config, name)
+    config_class =
+      begin
+        "Config::#{name.classify}".constantize
+      rescue
+        Config::Unsupported
+      end
```
Comment (gylaz) [inline app/models/config_builder.rb]: How about extracting this to a method to avoid this weird indentation?

File: spec/models/linter/base_spec.rb [hunk 2]
```diff
@@ -1,5 +1,4 @@
-require "spec_helper"
-require "app/models/linter/base"
+require "rails_helper"
```
Comment (gylaz) [inline spec/models/linter/base_spec.rb]: Why this change?

**Comment by gylaz** [comment]
:+1:

File: spec/models/linter/base_spec.rb (hunk 2 above)
Comment (teoljungberg) [inline spec/models/linter/base_spec.rb]: Since now we are requiring the ConfigBuilder, which in turn builds any of the `Config::*` objects. And I didn't want to require variations of that in this file.

Do you think we should?

Comment (teoljungberg) [inline spec/models/linter/base_spec.rb]: I changed this to require the needed files instead.
